lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb

```diff
@@ -1315,9 +1315,40 @@ def cmd_load(*args)
         if (client.core.use(modulenameprovided) == true)
           add_extension_client(md)
         end
-      rescue
+      rescue => ex
         print_line
-        log_error("Failed to load extension: #{$!}")
+        log_error("Failed to load extension: #{ex.message}")
+        if ex.kind_of?(ExtensionLoadError) && ex.name
+          # MetasploitPayloads and MetasploitPayloads::Mettle do things completely differently, build an array of
+          # suggestion keys (binary_suffixs and Mettle build-tuples)
+          suggestion_keys = MetasploitPayloads.list_meterpreter_extension_suffixs(ex.name) + MetasploitPayloads::Mettle.available_platforms(ex.name)
+          suggestion_map = {
+            'jar' => 'java',
+            'php' => 'php',
+            'py' => 'python',
+            'x64.dll' => 'windows/x64',
+            'x86.dll' => 'windows',
+            'mips64-linux-muslsf' => 'linux/mips64',
+            'mipsel-linux-muslsf' => 'linux/mipsle',
+            'powerpc64le-linux-musl' => 'linux/ppc64le',
+            'mips-linux-muslsf' => 'linux/mipsbe',
+            'powerpc-linux-muslsf' => 'linux/ppc',
+            's390x-linux-musl' => 'linux/zarch',
+            'x86_64-linux-musl' => 'linux/x64',
+            'i486-linux-musl' => 'linux/x86',
+            'armv5l-linux-musleabi' => 'linux/armle',
+            'aarch64-linux-musl' => 'linux/aarch64',
+            'armv5b-linux-musleabi' => 'linux/armbe',
+            'powerpc-e500v2-linux-musl' => 'linux/ppce500v2',
+          }
```

Comment:
I'm not a fan of this static mapping. Suggestions would be appreciated! The core of the issue is that `MetasploitPayloads` take [binary suffixes](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/base/sessions/meterpreter.rb#L626) while `MetasploitPayloads::Mettle` takes platforms (which are just build tuples like `x86_64-linux-musl`) to get the extensions code. In both cases, there is not an easy way to reverse the mapping to a session type (like [`Msf::Sessions::Meterpreter_x64_Win`](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/base/sessions/meterpreter_x64_win.rb).

A more flexible solution would need to be implemented for MetasploitPayloads where the platform and arch could yield the binary suffix. For Mettle, the build tuple would probably need to be defined as a class attribute and then all of the session types would need to be enumerated to identify the ones that are Mettle and then a payload would need to be inferred from them based on the platform / architecture.

Alternatively, this is likely the only place that it makes sense to have this level of suggestions since it's where the user explicitly their extensions.